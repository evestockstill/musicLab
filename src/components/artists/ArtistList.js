import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ArtistList = ({ artists }) => {
  const artistList = artists.map(artist => {
    return (
      <Link key={artist.id} to={`/albums/${artist.name}/${artist.id}`}>
        <li>{artist.name}</li>
      </Link>
    );
  });
  return (
    <ul>{artistList}</ul>
  );
};

ArtistList.propTypes = {
  artists: PropTypes.array.isRequired
};

export default ArtistList;
