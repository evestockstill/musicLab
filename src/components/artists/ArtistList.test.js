import React from 'react';
import { shallow } from 'enzyme';
import ArtistList from './ArtistList';

describe('ArtistList component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<ArtistList artists={[{ id: 1, name: 'test name' }, { id: 2, name: 'test name2' }]} match={{ params: { id: '1' }
    }} loading={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
