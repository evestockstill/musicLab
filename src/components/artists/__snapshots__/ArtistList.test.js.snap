// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList component matches a snapshot 1`] = `
<ul>
  <Link
    key="1"
    to="/albums/1"
  >
    <li>
      test name
    </li>
  </Link>
  <Link
    key="2"
    to="/albums/2"
  >
    <li>
      test name2
    </li>
  </Link>
</ul>
`;
